main/acle.md

```diff
@@ -9456,6 +9461,701 @@ are named after. All of the functions have external linkage.
+   svfloat32_t svdot[_f32_f16_f16](svfloat32_t zda, svfloat16_t zn, svfloat16_t zm);
+   svfloat32_t svdot_lane[_f32_f16_f16](svfloat32_t zda, svfloat16_t zn, svfloat16_t zm,
+                                uint64_t imm_idx);
+   ```
+
+#### EXTQ
+
+Extract vector segment from each pair of quadword segments.
+
+ ``` c
+   // Variants are also available for:
+   // _s8, _s16, _u16, _s32, _u32, _s64, _u64
+   // _bf16, _f16, _f32, _f64
+   svuint8_t svextq_lane[_u8](svuint8_t zdn, svuint8_t zm, uint64_t imm);
+   ```
+
+#### LD1D, LD1W
+
+Contiguous zero-extending load to quadword (single vector).
+
+ ``` c
+   // Variants are also available for:
+   // _u32, _s32
+   svfloat32_t svld1uwq[_f32](svbool_t, const float32_t *ptr);
+   svfloat32_t svld1uwq_vnum[_f32](svbool_t, const float32_t *ptr, int64_t vnum);
+  
+   // Variants are also available for:
+   // _u64, _s64
+   svfloat64_t svld1udq[_f64](svbool_t, const float64_t *ptr);
+   svfloat64_t svld1udq_vnum[_f64](svbool_t, const float64_t *ptr, int64_t vnum);
+ ```
+
+#### LD1B, LD1D, LD1H, LD1W
+
+Contiguous load to multi-vector.
+
+ ``` c
+   // All the intrinsics below are [SVE2.1 or SME2]
+   // Variants are also available for _s8
+   svuint8x2_t svld1[_u8]_x2(svcount_t png, const uint8_t *rn);
+   svuint8x4_t svld1[_u8]_x4(svcount_t png, const uint8_t *rn);
+   svuint8x2_t svld1_vnum[_u8]_x2(svcount_t png, const uint8_t *rn,
+                                  int64_t vnum);
+   svuint8x4_t svld1_vnum[_u8]_x4(svcount_t png, const uint8_t *rn,
+                                  int64_t vnum);
+
+
+   // Variants are also available for _s16, _f16 and _bf16
+   svuint16x2_t svld1[_u16]_x2(svcount_t png, const uint16_t *rn);
+   svuint16x4_t svld1[_u16]_x4(svcount_t png, const uint16_t *rn);
+   svuint16x2_t svld1_vnum[_u16]_x2(svcount_t png, const uint16_t *rn,
+                                    int64_t vnum);
+   svuint16x4_t svld1_vnum[_u16]_x4(svcount_t png, const uint16_t *rn,
+                                    int64_t vnum);
+
+
+   // Variants are also available for _s32 and _f32
+   svuint32x2_t svld1[_u32]_x2(svcount_t png, const uint32_t *rn);
+   svuint32x4_t svld1[_u32]_x4(svcount_t png, const uint32_t *rn);
+   svuint32x2_t svld1_vnum[_u32]_x2(svcount_t png, const uint32_t *rn,
+                                    int64_t vnum);
+   svuint32x4_t svld1_vnum[_u32]_x4(svcount_t png, const uint32_t *rn,
+                                    int64_t vnum);
+
+
+   // Variants are also available for _s64 and _f64
+   svuint64x2_t svld1[_u64]_x2(svcount_t png, const uint64_t *rn);
+   svuint64x4_t svld1[_u64]_x4(svcount_t png, const uint64_t *rn);
+   svuint64x2_t svld1_vnum[_u64]_x2(svcount_t png, const uint64_t *rn,
+                                    int64_t vnum);
+   svuint64x4_t svld1_vnum[_u64]_x4(svcount_t png, const uint64_t *rn,
+                                    int64_t vnum);
+   ```
+
+#### LD1Q
+
+Gather Load Quadword.
```

Comment:
There is only an unscaled variant of this instruction, so maybe don't have both offset and index?

Reply:
For the other SVE load and store intrinsics, we tried to provide a consistent interface and set of addressing modes.  So the deciding factor wasn't so much whether the call mapped to a single instruction, but whether the underlying instruction could easily emulate the mode.  “Single instruction” is a bit of nebulous concept anyway for loads and stores, since a single C address expression might need several operations to compute.

Since scaling is just a shift left, I think it's worth providing both index and offset variants.